deepchecks/nlp/utils/text_properties.py

```diff
@@ -260,14 +297,24 @@ def language(
     return language_codes
 
 
-def sentiment(raw_text: Sequence[str]) -> List[str]:
+def sentiment(raw_text: Sequence[str]) -> List[float]:
     """Return list of floats of sentiment."""
-    return [textblob.TextBlob(text).sentiment.polarity for text in raw_text]
+    if text_blob_cache.get('textblob') is None:
+        # TextBlob uses only the words and not the relations between them, so we can sample the text
+        # to speed up the process:
+        raw_text = [_sample_for_property(text=text, mode='words') for text in raw_text]
+        text_blob_cache['textblob'] = [textblob.TextBlob(text).sentiment for text in raw_text]
```

Comment:
what is the typical RAM size of this cache for a largish dataset?

Reply:
N * size(float). Though doesn't really matter, think that you're keeping it in memory anyway as 2 properties a few miliseconds later

Reply:
oh right thought it's a cache also for the strings

Reply:
Well there's one for that in the next PR, you're in for a treat. But it's technically limited.